src/Bundle/CoreBundle/Field/Types/ReferenceFieldType.php

```diff
@@ -302,4 +318,78 @@ function validateData(FieldableField $field, $data, ExecutionContextInterface $c
             }
         }
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    function validateSettings(FieldableFieldSettings $settings, ExecutionContextInterface $context)
+    {
+        // Validate allowed and required settings.
+        parent::validateSettings($settings, $context);
+
+        // Only continue, if there are no violations yet.
+        if ($context->getViolations()->count() > 0) {
+            return;
+        }
+
+        // Try to resolve content type. If it don't throw an exception, the domain and content_type exist and the user can access it.
+        try {
+            $this->resolveContentType($settings->domain, $settings->content_type);
+        }
+        catch (DomainAccessDeniedException $e) {
+            $context->buildViolation('invalid_domain')->atPath('domain')->addViolation();
+        } catch (MissingOrganizationException $e) {
+            $context->buildViolation('invalid_organization')->atPath('domain')->addViolation();
+        }
+
+
+        // Special case 1: We are validating a new domain, that is not already persisted.
+        catch (MissingDomainException $e) {
+
+            if($context->getRoot() instanceof Domain && empty($context->getRoot()->getId())) {
+
+                // If we don't reference the new domain, but the domain also does not exist, we can't reference it.
+                if($context->getRoot()->getIdentifier() != $settings->domain) {
+                    $context->buildViolation('invalid_domain')->atPath('domain')->addViolation();
+                    return;
+                }
+
+                $contentType = $context->getRoot()->getContentTypes()->filter(
+                    function (ContentType $contentType) use ($settings) {
+                        return $contentType->getIdentifier() == $settings->content_type;
+                    }
+                )->first();
+
+                // If we referenced content_type was not found in our new domain, we can't reference it.
+                if(!$contentType) {
+                    $context->buildViolation('invalid_content_type')->atPath('content_type')->addViolation();
+                    return;
+                }
+
+                return;
+            }
+
+            $context->buildViolation('invalid_domain')->atPath('domain')->addViolation();
+
+
+
+        // Special case 2: Domain does exist, but we are updating the domain at the moment, adding a new content_type.
+        } catch (MissingContentTypeException $e) {
+
+            if($context->getRoot() instanceof Domain && $this->uniteCMSManager->getDomain()  && $context->getRoot()->getId() === $this->uniteCMSManager->getDomain()->getId()) {
```

Comment:
Something is wrong here in this if statement "&& $this->uniteCMSManager->getDomain()  &&" does not make any sense